### PR TITLE
Disentagle `SORT_*`

### PR DESCRIPTION
**Pull request by flatcap**
In NeoMutt, all the sort methods, e.g. `SORT_EMAIL`, are defined centrally, in [`config/sort2.h`](https://github.com/neomutt/neomutt/blob/c391140ef109a8fdbbc53869cae5a40464a21013/config/sort2.h#L33-L60).
(this design is inherited from Mutt).
These constants are used throughout the program.

However, most of the users of the `SORT_*` constants are independent -- local to one Dialog or library.

This PR separates the constants reducing dependencies.

---

**Part 0: Prep**

This part clears the way for the library refactoring.
 
- Include-What-You-Use
- Rename the Email sort functions to `email_sort_*()`
- Move `sort.[ch]` to libemail
- Rename `config/sort2.h` to `config/sort.h`

---

**Part 1: Alias**

- Create Alias-specific sort methods
    ```
    ALIAS_SORT_ALIAS,       ///< Sort by Alias short name
    ALIAS_SORT_EMAIL,       ///< Sort by Email Address
    ALIAS_SORT_NAME,        ///< Sort by Real Name
    ALIAS_SORT_UNSORTED,    ///< Sort by the order the Aliases were configured
     ```
- Rename `$sort_alias` to `$alias_sort`
  (and create a synonym for backwards compatibility)
- Symbols dropped from `config/sort2.h`
   - `SORT_ALIAS`
   - `SORT_EMAIL`
   - `SORT_NAME`

---

**Part 2: Sidebar**

- Create Sidebar-specific sort methods
   ```
   SB_SORT_COUNT,       ///< Sort by total message count
   SB_SORT_DESC,        ///< Sort by mailbox description
   SB_SORT_FLAGGED,     ///< Sort by count of flagged messages
   SB_SORT_PATH,        ///< Sort by mailbox path (alphabetically)
   SB_SORT_UNREAD,      ///< Sort by count of unread messages
   SB_SORT_UNSORTED,    ///< Sort into the order the mailboxes were configured
   ```
- Rename `$sidebar_sort_method` to `$sidebar_sort`
  (and create a synonym for backwards compatibility)
- Symbols dropped from `config/sort2.h`
   - `SORT_FLAGGED`
   - `SORT_PATH`

---

**Part 3: Browser**

- Create Browser-specific sort methods
   ```
   BROWSER_SORT_ALPHA,      ///< Sort alphabetically by name
   BROWSER_SORT_COUNT,      ///< Sort by total message count
   BROWSER_SORT_DATE,       ///< Sort by date
   BROWSER_SORT_DESC,       ///< Sort by description
   BROWSER_SORT_NEW,        ///< Sort by count of new messages
   BROWSER_SORT_SIZE,       ///< Sort by size
   BROWSER_SORT_UNSORTED,   ///< Sort into the raw order
   ```
- Rename `$sort_browser` to `$browser_sort`
  (and create a synonym for backwards compatibility)
- Symbols dropped from `config/sort2.h`
   - `SORT_ALPHA`
   - `SORT_COUNT`
   - `SORT_DESC`
   - `SORT_UNREAD`

---

**Part 4: Ncrypt**

- Create Ncrypt-specific sort methods
   ```
   KEY_SORT_ADDRESS,    ///< Sort by address
   KEY_SORT_DATE,       ///< Sort by date
   KEY_SORT_KEYID,      ///< Sort by key id
   KEY_SORT_TRUST,      ///< Sort by trust level
   ```
- Rename `$pgp_sort_keys` to `$pgp_key_sort`
  (and create a synonym for backwards compatibility)
- Symbols dropped from `config/sort2.h`
   - `SORT_ADDRESS`
   - `SORT_KEYID`
   - `SORT_TRUST`

---

**Part 5: Email**

- Create Email-specific sort methods
   ```
   EMAIL_SORT_DATE,             ///< Sort by the date the email was sent
   EMAIL_SORT_DATE_RECEIVED,    ///< Sort by when the message were delivered locally
   EMAIL_SORT_FROM,             ///< Sort by the email's From field
   EMAIL_SORT_LABEL,            ///< Sort by the emails label
   EMAIL_SORT_SCORE,            ///< Sort by the email's score
   EMAIL_SORT_SIZE,             ///< Sort by the size of the email
   EMAIL_SORT_SPAM,             ///< Sort by the email's spam score
   EMAIL_SORT_SUBJECT,          ///< Sort by the email's subject
   EMAIL_SORT_THREADS,          ///< Sort by email threads
   EMAIL_SORT_TO,               ///< Sort by the email's To field
   EMAIL_SORT_UNSORTED,         ///< Sort by the order the messages appear in the mailbox
   ```
- Symbols dropped from `config/sort2.h`
- `SORT_DATE`
- `SORT_FROM`
- `SORT_LABEL`
- `SORT_ORDER`
- `SORT_RECEIVED`
- `SORT_SCORE`
- `SORT_SIZE`
- `SORT_SPAM`
- `SORT_SUBJECT`
- `SORT_THREADS`
- `SORT_TO`

### CODE REVIEW

File: ChangeLog.md [hunk 1]
```diff
@@ -861,7 +861,7 @@
   - 100% Czech
   - 70% Turkish
 * Docs
-  - Document that `$sort_alias` affects the query menu
+  - Document that `$alias_sort` affects the query menu
```
Comment (roccoblues) [inline ChangeLog.md:864]: nit: I wouldn't change past ChangeLog entries.

File: alias/functions.c [hunk 2]
```diff
@@ -50,6 +50,7 @@
 #include "alias.h"
 #include "functions.h"
 #include "gui.h"
+#include "sort3.h"
```
Comment (roccoblues) [inline alias/functions.c:53]: Why is this file named `sort3.h` and not `sort.h`?

File: alias/functions.c [hunk 3]
```diff
@@ -405,19 +406,19 @@ static int op_sort(struct AliasMenuData *mdata, int op)
...
       break;
 
     case 3: /* (e)mail */
-      sort = SORT_EMAIL;
+      sort = ALIAS_SORT_EMAIL;
       break;
 
     case 4: /* (u)nsorted */
-      sort = SORT_ORDER;
+      sort = ALIAS_SORT_UNSORTED;
```
Comment (roccoblues) [inline alias/functions.c:421]: ❤️ calling unsorted `SORT_ORDER` is confusing.

File: browser/config.c [hunk 4]
```diff
@@ -35,20 +35,22 @@
 #include "config/lib.h"
 #include "lib.h"
 #include "expando/lib.h"
+#include "sort5.h"
```
Comment (roccoblues) [inline browser/config.c:38]: Ok, it's getting weirder. 😄  Not we are at `5`. Why?

File: ncrypt/config.c [hunk 5]
```diff
@@ -39,17 +39,18 @@
 #include "pgp.h"
 #include "pgplib.h"
 #include "smime.h"
+#include "sort6.h"
```
Comment (roccoblues) [inline ncrypt/config.c:42]: You know what I'm going to ask right?

**Review by roccoblues — APPROVED** [review]
LGTM only the header file names surprise me

**Comment by flatcap** [comment]
hehe, yeah the sort naming is a hilarious, but hopefully temporary 
there's `/sort.h` and `/config/sort2.h` that quite a few people need 
then each library needed its own  
now that I know where I'm going (no jokes, please :-), I can rename / rearrange the first two,
so that each library just has a private sort.h (unnumbered) 
I _think_ :-)

**Comment by flatcap** [comment]
Complete!
There are now 6 different `sort.h`s -- each private to a different library.
`SORT_ORDER` is dead, long live `SORT_UNSORTED` :-)

**Pull request merged**